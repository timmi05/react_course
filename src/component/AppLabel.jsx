import React from 'react';
import './AppLabel.css';

export const AppLabel = () =>
    <div className='app-label'>
        <span className='app-label-start'>netflix</span>
        <span className='app-label-end'>roulette</span>
    </div>;
