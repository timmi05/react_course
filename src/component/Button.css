.button{
  background-color: #f65261;
  color: #ffffff;
}

.button:disabled{
background-color: #555555
}

