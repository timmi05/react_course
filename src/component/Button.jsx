import React from 'react';
import './button.css';
import classnames from 'classnames';

const Button = props => {

    const { label, classes, handleSubmit, disabled } = props;
    const composedClassName = classnames(
        'button',
        classes,
      );

    return (
    <button className={ composedClassName } onClick={ handleSubmit } type="submit"
     disabled={ disabled }>{ label }</button>
    );
};
export default Button
