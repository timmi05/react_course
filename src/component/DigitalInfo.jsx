import React from 'react';

const DigitalInfo = props => {
    const { value, meaning } = props;
    return (
         <span className='digital-info'>
            <span className='digital-info-value'>{ value }</span>
            <span className='digital-info-meaning'>{ meaning }</span>
        </span>
    );
};

export default DigitalInfo