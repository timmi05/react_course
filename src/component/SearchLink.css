.search-link{
    color: #f65261;
}