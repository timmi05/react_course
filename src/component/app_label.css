.app-label{
    color: #f65261;
    font-size: 24px;
}

.app-label-start{
    font-weight: bold;
}

.app-label-end{

}