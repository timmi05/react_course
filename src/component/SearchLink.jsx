import React from 'react';
import './SearchLink.css';

export const SearchLink = props => {
    const { productSelected } = props;

    if ( !productSelected ) {
        return (
            <a/>
        );
    }
    return (
            <a  className='search-link' href=''>search</a>
        )
};