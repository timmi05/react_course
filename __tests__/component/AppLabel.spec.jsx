import React from 'react';
import { shallow } from 'enzyme';
import { AppLabel } from '../../src/component/AppLabel';

describe('AppLabel rendering', () => {
    it('renders correctly', () => {
        const component = shallow(<AppLabel/>);
        expect(component).toMatchSnapshot();
    });
});
