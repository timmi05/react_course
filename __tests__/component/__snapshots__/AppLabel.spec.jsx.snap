// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLabel rendering renders correctly 1`] = `
<div
  className="app-label"
>
  <span
    className="app-label-start"
  >
    netflix
  </span>
  <span
    className="app-label-end"
  >
    roulette
  </span>
</div>
`;
